### ISSUE TITLE
Seems Like Unpack is not Picklable

### ISSUE THREAD

**Issue by rizar** [bartvm/fuel#44]
My first attempt to use it ended up with the message below. I could not immediately find the reason in the code, will look later.

```
cPickle.PicklingError: Pickling failed.

Blocks relies on the ability to pickle the entire main loop, which includes all bricks, data streams, extensions, etc. All of these must be serializable using Pythons pickling library. This means certain things such as nested functions, lambda expressions, generators, etc. should be avoided. Please see the documentation for more detail.

Original exception:
        PicklingError: Can't pickle <type 'iterator'>: attribute lookup __builtin__.iterator failed

Original exception:
        PicklingError: Pickling failed.

Blocks relies on the ability to pickle the entire main loop, which includes all bricks, data streams, extensions, etc. All of these must be serializable using Pythons pickling library. This means certain things such as nested functions, lambda expressions, generators, etc. should be avoided. Please see the documentation for more detail
```

**Comment by rizar** [comment]
Localized: this happens when `numpy.array` is unpacked, because `picklable_itertools` does not know about such an object, that is it's `iter_` method simply calls global `iter`. 

@dwf, do you think we should teach `picklable_itertools` to handle numpy arrays?

**Comment by dwf** [comment]
I'd say yes, with the proviso that we should not make NumPy an absolute
dependency.

**Comment by dwf** [comment]
Fixed in dwf/picklable_itertools@825eb0f. @rizar please confirm it works with Unpack (someone might want to write a test, so I opened #46, which you can feel free to tackle or not).

**Comment by rizar** [comment]
I confirm that it works, covering it with a regression test can wait in the CCW queue a bit, I guess :) This was not Fuel's fault anyway.
